Sources/Datadog/DatadogInternal/Upload/URLRequestBuilder.swift

```diff
@@ -100,27 +111,31 @@ import Foundation
         self.headers = headers
     }
 
-    /// Creates `URLRequest` for uploading given `data` to Datadog.
-    /// - Parameter data: data to be uploaded
-    /// - Returns: Returns: the `URLRequest` object.
-    func uploadRequest(with data: Data) -> URLRequest {
+    /// Creates `URLRequest` for uploading given `body` to Datadog.
+    ///
+    /// - Parameter body: HTTP body to be attached to request
+    /// - Parameter compress: if `body` should be compressed into ZLIB Compressed Data Format (IETF RFC 1950)
+    /// - Returns: the `URLRequest` object.
+    public func uploadRequest(with body: Data, compress: Bool = true) -> URLRequest {
```

Comment:
Adding `Bool` argument is not a perfect way, but unlocks me to move forward with introducing limited complexity to the SDK. We could think of refactoring this later, e.g. making `RequestBuilder` only prepare pre-configured `URLRequest` that Features set `.httpBody` on in their own routines.

Reply:
I agree with that, tho we might need custom headers sometimes. This could return a custom object instead of a `URLRequest`. Will see when we refactor the upload 👍 

In which condition SR can decide to not compress the data?

Reply:
> In which condition SR can decide to not compress the data?

All the time - SR does its own compression before making this call. It always sets `compress: false` and passes _"zlib w/ flush sync"_ compressed `data` as `body` to `uploadRequest(with:compress:)`.